pkg/controller/flightctl/flightctl.go

```diff
@@ -0,0 +1,186 @@
+	IsManagedClusterAFlightctlDevice(ctx context.Context, managedClusterName string) (bool, error)
+}
+
+func NewFlightCtl(clientHolder *helpers.ClientHolder, clusterIngressDomain string) FlightCtler {
+	return &FlightCtl{
+		flightctlServer:         FlightCtlInternalServer,
+		agentRegistrationServer: "https://agent-registration-multicluster-engine." + clusterIngressDomain,
+		clientHolder:            clientHolder,
+	}
+}
+
+type FlightCtl struct {
+	clientHolder            *helpers.ClientHolder
+	flightctlServer         string
+	agentRegistrationServer string
+	cachedToken             string
+	cachedCA                string
+}
+
+var _ FlightCtler = &FlightCtl{}
+
+func (f *FlightCtl) IsFlightCtlEnabled(ctx context.Context) (bool, error) {
+	// If `flightctl` ns is not created, means the flightctl is not enabled.
+	ns := &corev1.Namespace{}
+	err := f.clientHolder.RuntimeClient.Get(ctx, types.NamespacedName{
+		Name: FlightCtlNamespace,
+	}, ns)
+
+	if errors.IsNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (f *FlightCtl) ApplyRepository(ctx context.Context) error {
+	token, err := f.getFlightCtlClientServiceAccountToken(ctx)
+	if err != nil {
+		return err
+	}
+
+	ca, err := f.getAgentRegistrationCA(ctx)
+	if err != nil {
+		return err
+	}
+
+	expectedRepository := &flightctlapiv1.Repository{
+		ApiVersion: "v1alpha1",
+		Kind:       "Repository",
+		Metadata: flightctlapiv1.ObjectMeta{
+			// Note: In the fleets' `httpRef.repository` field, the name is `acm-registration`.
+			// See details in: https://github.com/flightctl/flightctl/blob/main/docs/user/registering-microshift-devices-acm.md
+			Name: ptr("acm-registration"),
+		},
+		Spec: flightctlapiv1.RepositorySpec{},
+	}
+	expectedRepository.Spec.MergeHttpRepoSpec(flightctlapiv1.HttpRepoSpec{
+		Type: flightctlapiv1.Http,
+		Url:  f.agentRegistrationServer,
+		HttpConfig: flightctlapiv1.HttpConfig{
+			Token: &token,
+			CaCrt: &ca,
+		},
+		ValidationSuffix: ptr("/agent-registration"),
+	})
+
+	return flightctlcli.ApplyRepository(ctx, token, f.flightctlServer, expectedRepository)
+}
+
+func (f *FlightCtl) IsManagedClusterAFlightctlDevice(ctx context.Context, managedClusterName string) (bool, error) {
+	token, err := f.getFlightCtlClientServiceAccountToken(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	response, err := flightctlcli.GetDevice(ctx, token, f.flightctlServer, managedClusterName)
+	if err != nil {
+		return false, err
+	}
+
+	if response.HTTPResponse.StatusCode == http.StatusNotFound {
+		return false, nil
+	}
+
+	if response.HTTPResponse.StatusCode != http.StatusOK {
+		return false, fmt.Errorf("failed to get device %s, status code: %d", managedClusterName, response.HTTPResponse.StatusCode)
+	}
+
+	return true, nil
+}
+
+func (f *FlightCtl) getFlightCtlClientServiceAccountToken(ctx context.Context) (string, error) {
+	if f.cachedToken != "" {
+		return f.cachedToken, nil
+	}
+
+	secret := &corev1.Secret{}
```

Comment:
you should use tokenRequest instead of secret. The secret way is not secure and will be disabled by default in the future.

Reply:
Updated to tokenRequest way.